lib/DDG/Meta/Data.pm

```diff
@@ -36,91 +34,75 @@ unless(%ia_metadata){
         mkdir $mdir or die "Failed to mkdir $mdir: $!";
     }
 
-    # Load IA metadata. Not all types are required during development.
-    for my $iat (@ia_types){
-        debug && warn "Processing IA type: $iat";
-
-        my $json_endpt = lc $iat;
-        $json_endpt =~ s/goodie/goodies/;
+    debug && warn "Processing metadata";
 
-        # Prefer freshly downloaded metadata and fall back to metadata
-        # bundled with installed IA repos
-        my $f = "$mdir/$iat.json";
-        my $c = $ENV{NO_METADATA_DOWNLOAD} ? 0 : mirror("https://duck.co/ia/repo/$json_endpt/json", $f);
-        my $json;
-        if($c == RC_OK || $c == RC_NOT_MODIFIED){
-            $json = io($f)->slurp; 
+    my $f = "$mdir/metadata.json";
+    unless($ENV{NO_METADATA_DOWNLOAD}){
+        my $ua = LWP::UserAgent->new;
+        $ua->timeout(5);
+        $ua->default_header('Accept-Encoding' => scalar HTTP::Message::decodable());
+        my $res = $ua->mirror("https://duck.co/ia/repo/all/json", $f);
```

Comment:
No - https://metacpan.org/source/ETHER/libwww-perl-6.13/lib/LWP/UserAgent.pm#L929.  Download is saved to a temp file that is only swapped with the existing after several checks have passed.